src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java

```diff
@@ -0,0 +1,131 @@
+
+    public UserAPI(Main main) {
+        super(main, RECIPE_ID.PASSWORDLESS.toString());
+    }
+
+    @Override
+    public String getPath() {
+        return "/recipe/user";
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        // logic based on: https://app.code2flow.com/flowcharts/617a9aafdc97ee415448db74
+        String userId = InputParser.getQueryParamOrThrowError(req, "userId", true);
+        String email = InputParser.getQueryParamOrThrowError(req, "email", true);
+        String phoneNumber = InputParser.getQueryParamOrThrowError(req, "phoneNumber", true);
+
+        if (Stream.of(userId, email, phoneNumber).filter(Objects::nonNull).count() != 1) {
+            throw new ServletException(
+                    new BadRequestException("Please provide exactly one of userId, email or phoneNumber"));
+        }
+
+        try {
+            UserInfo user;
+            if (userId != null) {
+                user = Passwordless.getUserById(main, userId);
+            } else if (email != null) {
+                user = Passwordless.getUserByEmail(main, Utils.normaliseEmail(email));
+            } else {
+                user = Passwordless.getUserByPhoneNumber(main, phoneNumber);
+            }
+
+            if (user == null) {
+                JsonObject result = new JsonObject();
+                result.addProperty("status", userId != null ? "UNKNOWN_USER_ID_ERROR"
+                        : email != null ? "UNKNOWN_EMAIL_ERROR" : "UNKNOWN_PHONE_NUMBER_ERROR");
+                super.sendJsonResponse(200, result, resp);
+            } else {
+                JsonObject result = new JsonObject();
+                result.addProperty("status", "OK");
+
+                JsonObject userJson = new JsonParser().parse(new Gson().toJson(user)).getAsJsonObject();
+                result.add("user", userJson);
+                super.sendJsonResponse(200, result, resp);
+            }
+        } catch (StorageQueryException e) {
+            throw new ServletException(e);
+        }
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        // logic based on: https://app.code2flow.com/TXloWHJOwWKg
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        String userId = InputParser.parseStringOrThrowError(input, "userId", false);
+
+        FieldUpdate emailUpdate = !input.has("email") ? null
+                : new FieldUpdate(input.get("email").isJsonNull() ? null
+                        : Utils.normaliseEmail(InputParser.parseStringOrThrowError(input, "email", false)));
+
+        FieldUpdate phoneNumberUpdate = !input.has("phoneNumber") ? null
+                : new FieldUpdate(input.get("phoneNumber").isJsonNull() ? null
+                        : InputParser.parseStringOrThrowError(input, "phoneNumber", false));
+
```

Comment:
If someone doesn't pass email and phoneNumber, then we must throw a bad request?

Reply:
It's a no-op, so I'm not sure if we need to throw: the request is not to update either, so we don't.

Reply:
Sounds good.